mllib/src/test/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifierSuite.scala

```diff
@@ -82,6 +83,23 @@ class MultilayerPerceptronClassifierSuite
     }
   }
 
+  test("test model probability") {
+    val layers = Array[Int](2, 5, 2)
+    val trainer = new MultilayerPerceptronClassifier()
+      .setLayers(layers)
+      .setBlockSize(1)
+      .setSeed(123L)
+      .setMaxIter(100)
+      .setSolver("l-bfgs")
+    val model = trainer.fit(dataset)
+    model.setProbabilityCol("probability")
+    val result = model.transform(dataset)
+    val features2prob = udf { features: Vector => model.mlpModel.predict(features) }
+    val cmpVec = udf { (v1: Vector, v2: Vector) => v1 ~== v2 relTol 1e-3 }
+    assert(result.select(cmpVec(features2prob(col("features")), col("probability")))
+      .rdd.map(_.getBoolean(0)).reduce(_ && _))
+  }
+
```

Comment:
I think we should include a stronger test for this. I did a quick search and couldn't find a strong test for `mlpModel.predict`, it might be good to add one. Also, I believe this xor dataset only produces probability predictions ~equal to 0 or 1.